### PR TITLE
update IBFT2.0 transitions

### LINKED ISSUE #843: IBFT 2.0 block rewards are configurable using transitions
See: https://github.com/hyperledger/besu/pull/1977

### PR DESCRIPTION
**Pull request by rolandtyler**
Signed-off-by: Roland Tyler <roland.tyler@consensys.net>

## Describe the change
Edit transitions for IBFT2.0:
- update block period
- add block reward

## Issue fixed
Fixes #843 

### For content changes

- [x] Doc content
- [ ] Doc pages organisation

### For tools changes

- [ ] CircleCI workflow
- [ ] Build and QA tools (lint, vale,…)
- [ ] MkDocs templates
- [ ] MkDocs configuration
- [ ] Python dependencies
- [ ] Node dependencies and JavaScript
- [ ] ReadTheDocs configuration
- [ ] GitHub integration

## Testing
https://hyperledger-besu--882.org.readthedocs.build/en/882/HowTo/Configure/Consensus-Protocols/IBFT/

### CODE REVIEW

File: docs/HowTo/Configure/Consensus-Protocols/IBFT.md [hunk 1]
```diff
@@ -181,7 +180,22 @@ To tune the block timeout for your network deployment:
     View [`TRACE` logs](../../../Reference/API-Methods.md#admin_changeloglevel) to see round change
     log messages.
 
-#### Configure block time on an existing network deployment
+Use a [transition](#transitions) to update the `blockperiodseconds` in an existing network.
+
+{!global/Config-Options.md!}
+
+## Transitions
```
Comment (siladu) [inline docs/HowTo/Configure/Consensus-Protocols/IBFT.md:187]: I like that Transitions have their own section now, that'll be helpful for linking to it which I seem to be doing more often these days :)

File: docs/HowTo/Configure/Consensus-Protocols/IBFT.md [hunk 2]
```diff
@@ -245,12 +259,87 @@ To update an existing network with a new `blockperiodseconds`:
...
+        === "Syntax"
+
+            ```bash
+            {
+              "config": {
+                 ...
+                 "ibft2": {
+                   "blockperiodseconds": 2,
+                   "epochlength": 30000,
+                   "requesttimeoutseconds": 4
```
Comment (siladu) [inline docs/HowTo/Configure/Consensus-Protocols/IBFT.md:283]: Maybe should have an initial blockreward value here to show what its being changed in the transition below?

File: docs/HowTo/Configure/Consensus-Protocols/IBFT.md [hunk 3]
```diff
@@ -245,12 +259,87 @@ To update an existing network with a new `blockperiodseconds`:
...
+        === "Example"
+
+            ```bash
+            {
+              "config": {
+                 ...
+                 "ibft2": {
+                   "blockperiodseconds": 2,
+                   "epochlength": 30000,
+                   "requesttimeoutseconds": 4
```
Comment (siladu) [inline docs/HowTo/Configure/Consensus-Protocols/IBFT.md:315]: As above, initial blockreward would be good here, unless the example intends to show 0 -> +ve value

File: docs/HowTo/Configure/Consensus-Protocols/IBFT.md [hunk 4]
```diff
@@ -245,12 +259,87 @@ To update an existing network with a new `blockperiodseconds`:
...
+                 },
+                 "transitions": {
+                   "ibft2": [
+                   {
+                     "block": 10,
+                     "blockreward": "1"
```
Comment (siladu) [inline docs/HowTo/Configure/Consensus-Protocols/IBFT.md:321]: The blockreward unit is Wei. If the aim of these examples is to show realistic number, we could probably add quite a few more 0s on the end :)

Comment (siladu) [reply docs/HowTo/Configure/Consensus-Protocols/IBFT.md:321]: e.g. https://besu.hyperledger.org/en/stable/HowTo/Configure/Consensus-Protocols/IBFT/#genesis-file uses "5000000000000000" Wei

File: docs/HowTo/Use-Privacy/Performance-Best-Practices.md [hunk 5]
```diff
@@ -14,6 +14,7 @@ However the following item is of such outstanding impact to performance that it
 
 Running EVM transactions creates a lot of random reads that are executed sequentially.
 The Besu data folder should be located on the fastest possible storage media.
+
```
Comment (siladu) [inline docs/HowTo/Use-Privacy/Performance-Best-Practices.md:17]: errant space?

File: docs/HowTo/Configure/Consensus-Protocols/IBFT.md (hunk 2 above)
Comment (siladu) [reply docs/HowTo/Configure/Consensus-Protocols/IBFT.md:283]: It defaults to zero if you don't specify it in the initial config, so the example is still correct. If the intention is to go from 0 -> +ve reward then IMO we should make that explicit either in the preamble or by specifying initial config.

**Review by siladu — CHANGES_REQUESTED** [review]
blockreward transitions apply to QBFT as well so it could be added to here https://besu.hyperledger.org/en/latest/HowTo/Configure/Consensus-Protocols/QBFT/#transitions
Not sure if that's beyond the scope of this PR though :)

A few minor points in the comments...

File: docs/HowTo/Use-Privacy/Performance-Best-Practices.md (hunk 5 above)
Comment (siladu) [reply docs/HowTo/Use-Privacy/Performance-Best-Practices.md:17]: ah...just realised that was the markdown fix 🏅

**Comment by rolandtyler** [comment]
> blockreward transitions apply to QBFT as well so it could be added to here https://besu.hyperledger.org/en/latest/HowTo/Configure/Consensus-Protocols/QBFT/#transitions Not sure if that's beyond the scope of this PR though :)
> 
> A few minor points in the comments...

@siladu, thanks for the feedback.  I updated the examples and added the QBFT transition, as well as `miningbeneficiary`.    Could you re-review? Thanks!

File: docs/HowTo/Configure/Consensus-Protocols/IBFT.md [hunk 6]
```diff
@@ -241,16 +253,98 @@ To update an existing network with a new `blockperiodseconds`:
...
             ```
 
-1. Restart all nodes in the network using the updated genesis file.
-1. To verify the changes after the transition block, call
+3. Restart all nodes in the network using the updated genesis file.
+4. To verify the changes after the transition block, call
    [`ibft_getValidatorsByBlockNumber`](../../../Reference/API-Methods.md#ibft_getvalidatorsbyblocknumber), specifying `latest`.
 
-!!! caution
+### Update block rewards on an existing network deployment
```
Comment (siladu) [inline docs/HowTo/Configure/Consensus-Protocols/IBFT.md:260]: The block period uses the title "Configure block time...". Should these be consisent, i.e. either both use "Configure..." or both use "Update..."?

File: docs/HowTo/Configure/Consensus-Protocols/QBFT.md [hunk 7]
```diff
@@ -405,6 +402,95 @@ To update an existing network with a new `blockperiodseconds`:
 4. To verify the changes after the transition block, call
    [`qbft_getValidatorsByBlockNumber`](../../../Reference/API-Methods.md#ibft_getvalidatorsbyblocknumber), specifying `latest`.
 
+### Update block rewards on an existing network deployment
```
Comment (siladu) [inline docs/HowTo/Configure/Consensus-Protocols/QBFT.md:405]: Same point about "Configure..." and "Update..."

**Review by siladu — COMMENTED** [review]
Nice one for adding to QBFT as well :)

I'm not sure why there are changes to the `miningbeneficiary` ...I don't believe it's possible to update that one, but apologies if I missed something - where did that requirement come from?

**Review by siladu — APPROVED** [review]
LGTM :)

File: docs/HowTo/Configure/Consensus-Protocols/IBFT.md [hunk 8]
```diff
@@ -241,16 +255,93 @@ To update an existing network with a new `blockperiodseconds`:
             }
             ```
 
-1. Restart all nodes in the network using the updated genesis file.
-1. To verify the changes after the transition block, call
+3. Restart all nodes in the network using the updated genesis file.
```
Comment (bgravenorst) [inline docs/HowTo/Configure/Consensus-Protocols/IBFT.md:258]: @rolandtyler these get automatically numbered if you leave all the ordered list items as `1.`

**Review by bgravenorst — APPROVED** [review]
Just the one comment. otherwise, LGTM.

File: docs/HowTo/Configure/Consensus-Protocols/IBFT.md (hunk 8 above)
Comment (rolandtyler) [reply docs/HowTo/Configure/Consensus-Protocols/IBFT.md:258]: The version of IntelliJ I have has some auto feature that automatically renumbers them when I enter them as `1`.  There's still no release yet to fix it.  [Issue link](https://youtrack.jetbrains.com/issue/IDEA-273049).

**Pull request merged**